### PR TITLE
feat(pension supplement): new application

### PR DESCRIPTION
**Pull request by karenbjorg**
# Application for Pension Supplement

## What

New application for Social Insurance Administration (TR) where applicant can apply for pension supplement.

umsoknir/uppbot-a-lifeyri/

### Gervimenn to test the flow of the application
María ÞÍ Lulic - 0111439

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by ylfahfa — APPROVED** [review]
lgtm

File: libs/application/templates/social-insurance-administration/pension-supplement/src/lib/answerValidationSections/fileUploadValidationSection.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { Answer, Application } from '@island.is/application/types'
+import isEmpty from 'lodash/isEmpty'
+import { AnswerValidationConstants, ApplicationReason } from '../constants'
+import { getApplicationAnswers } from '../pensionSupplementUtils'
+import { buildError } from './utils'
+import { errorMessages } from '@island.is/application/templates/social-insurance-administration-core/lib/messages'
+
+export const fileUploadValidationSection = (
```
Comment (Thorrik58) [inline libs/application/templates/social-insurance-administration/pension-supplement/src/lib/answerValidationSections/fileUploadValidationSection.ts:8]: This should be avoided and should be possible with a dataschema validation.

File: libs/application/templates/social-insurance-administration/pension-supplement/src/lib/PensionSupplementTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+                ),
+            },
+          ],
+        },
+      },
+    },
+  },
+  stateMachineOptions: {
+    actions: {
+      setApproveExternalData: assign((context) => {
```
Comment (Thorrik58) [inline libs/application/templates/social-insurance-administration/pension-supplement/src/lib/PensionSupplementTemplate.ts:344]: Why are you doing this? The input should take care of marking the approve external data and the dataschema should validate that the user has approved it.

File: libs/application/templates/social-insurance-administration/pension-supplement/src/lib/PensionSupplementTemplate.ts [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+        const { answers } = application
+
+        set(answers, 'approveExternalData', true)
+        return context
+      }),
+      /**
+       * Copy the current answers to temp. If the user cancels the edits,
+       * we will restore the answers to their original state from temp.
+       */
+      createTempAnswers: assign((context, event) => {
```
Comment (Thorrik58) [inline libs/application/templates/social-insurance-administration/pension-supplement/src/lib/PensionSupplementTemplate.ts:355]: Why is this needed? Dont the answers get assigned to the application object?

**Review by Thorrik58 — COMMENTED** [review]
Just come questions

File: libs/application/templates/social-insurance-administration/pension-supplement/src/lib/PensionSupplementTemplate.ts (hunk 3 above)
Comment (Thorrik58) [reply libs/application/templates/social-insurance-administration/pension-supplement/src/lib/PensionSupplementTemplate.ts:347]: @karenbjorg im happy to approve but I would love an answer to why this is needed?

**Review by saevarma — APPROVED** [review]
Core files LGTM

**Review by Thorrik58 — APPROVED** [review]
